karpenter/add-karpenter-donot-disrupt/add-karpenter-donot-disrupt.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: add-karpenter-donot-disrupt
+  annotations:
+    policies.kyverno.io/title: Add Karpenter Do Not Disrupt
+    policies.kyverno.io/category: Karpenter, EKS Best Practices
+    policies.kyverno.io/severity: medium
+    policies.kyverno.io/subject: Pod
+    kyverno.io/kyverno-version: 1.7.1
+    policies.kyverno.io/minversion: 1.6.0
+    kyverno.io/kubernetes-version: "1.23"
+    policies.kyverno.io/description: >-
+      This policy assists in migrating from Karpenter 0.x to 1.x by adding the
+      karpenter.sh/do-not-disrupt label to pods that have the deprecated
+      karpenter.sh/do-not-evict label. This ensures smooth upgrades and
+      maintains the intended protection against pod disruption.
+spec:
+  rules:
+    - name: add-donot-disrupt-label
+      match:
+        resources:
+          kinds:
+            - Pod
```

Comment:
The original policy targets Job and CronJob, do we want the same?

Reply:
Thank you for reviewing

The original and this new policy have different purposes:
1. Original policy (add-karpenter-donot-evict) adds the evict annotation to all **Jobs/CronJobs**
2. This new policy (add-karpenter-donot-disrupt) is specifically for **migration** purposes - it adds the new disrupt label when it finds the old evict label, as mentioned in the Karpenter migration guide.

Since this policy is meant to assist in migration from 0.x to 1.x, it currently targets Pods to catch all cases where the old label exists. However, if you think we should maintain consistency with the original policy and target Jobs/CronJobs instead, I can modify it accordingly.

Reply:
Thanks for the clarification! If it's only for migration purpose then yes, targeting pods sounds good!